### DISCUSSION TITLE
autocomplete always adds $0 at the end when using mini.completion

### DISCUSSION THREAD

**Discussion by olekspickle** [mrcjkb/rustaceanvim#394] [Q&A] [answered]
### Have you read the docs and searched existing issues?

- [X] I have read [the README's troubleshooting guide](https://github.com/mrcjkb/rustaceanvim/blob/master/README.md#stethoscope-troubleshooting).
- [X] I have searched [existing issues](https://github.com/mrcjkb/rustaceanvim/issues?q=is%3Aopen+is%3Aissue+label%3Abug).
- [X] I have searched [existing discussions](https://github.com/mrcjkb/rustaceanvim/discussions).
- [X] I have searched the existing issues of other plugins related to this issue (if applicable).
- [X] I have searched the [existing rust-analyzer issues](https://github.com/rust-lang/rust-analyzer/issues?q=is%3Aissue+) (if applicable).
- [X] I can reproduce the issue with stable Neovim (not a distribution or nightly) and the [minimal config](https://github.com/mrcjkb/rustaceanvim/blob/master/troubleshooting/minimal.lua).

### Neovim version (nvim -v)

v0.9.1

### Operating system/version

Ubuntu 23.10

### Output of :checkhealth rustaceanvim

```console
rustaceanvim: require("rustaceanvim.health").check()

Checking for Lua dependencies ~
- OK [mfussenegger/nvim-dap](https://github.com/mfussenegger/nvim-dap) installed.

Checking external dependencies ~
- OK rust-analyzer: found rust-analyzer 1.78.0 (9b00956 2024-04-29)
- OK Cargo: found cargo 1.78.0 (54d8815d0 2024-03-26)
- OK rustc: found rustc 1.78.0 (9b00956e5 2024-04-29)

Checking config ~
- OK No errors found in config.

Checking for conflicting plugins ~
- OK No conflicting plugins detected.

Checking for tree-sitter parser ~
- OK tree-sitter parser for Rust detected.
```

### How to reproduce the issue

```console
require("mini.completion").setup{}

# and then just try any function with completion

fn main() {
 let var = std::env::*Tab*
}
```

### Expected behaviour

fn main() {
let var = std::env::var()
}

### Actual behaviour

fn main() {
let var = std::env::var(${1:key})$0
}

### The minimal config used to reproduce this issue.

```Lua
-- [[ lazy.lua ]]
return require('lazy').setup({
    {
        -- Awesome lightweight overall life improvement collection of plugins
        { 'echasnovski/mini.nvim', version = false },
        -- Rust support
        { 'mrcjkb/rustaceanvim', version = '^4',
            lazy = false, -- This plugin is already lazy
        }
})

require("mini.completion").setup{}
-- and minimal rustacean config
```

**Comment by olekspickle** [comment]
I did create an issue with mini.completion first, but they [said it is lsp fault](https://github.com/echasnovski/mini.nvim/issues/882#issuecomment-2114701265), and I experience this only with rust LSP

I'm not even sure it's even a bug or am I just stoopid and don't know how LSPs work in general, but any help would be appreciated.

**Comment by mrcjkb** [comment, marked as answer]
Hey :wave: 

this plugin doesn't do anything related to autocompletion, but it does register  the capablilty `textDocument.completion.completionItem.snippetSupport = true` by default.
Most completion plugins have snippet support, and Neovim 0.10 (nightly) has native snippet support.
So this feels like a sane default to me.

mini.completion doesn't have snippet support, so you need to disable it, by setting

```lua
vim.g.rustaceanvim.server.capabilities.textDocument.completion.completionItem.snippetSupport = false
```

See also: `:h rustaceanvim.config`.

I will push a hotfix to make sure rustaceanvim only enables snippet support for Neovim >= 0.10.

**Reply by mrcjkb** [reply]
Just pushed the fix.

**Comment by olekspickle** [comment]
thank you for quick response and explanation!
